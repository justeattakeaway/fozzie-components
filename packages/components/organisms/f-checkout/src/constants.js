export const CHECKOUT_METHOD_COLLECTION = 'collection';
export const CHECKOUT_METHOD_DELIVERY = 'delivery';

export const VALID_CHECKOUT_METHODS = [CHECKOUT_METHOD_COLLECTION, CHECKOUT_METHOD_DELIVERY];

export const TENANT_MAP = {
    'en-GB': 'uk',
    'en-AU': 'au',
    'en-NZ': 'nz',
    'da-DK': 'dk',
    'es-ES': 'es',
    'en-IE': 'ie',
    'it-IT': 'it',
    'nb-NO': 'no'
};

export const VALIDATIONS = {
    address: 'addressValidations',
    guest: 'guestValidations'
};
