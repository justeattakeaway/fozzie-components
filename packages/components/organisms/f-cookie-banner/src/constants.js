export const CONSENT_COOKIE_NAME = 'je-cookieConsent';
export const LEGACY_CONSENT_COOKIE_NAME = 'je-banner_cookie';
