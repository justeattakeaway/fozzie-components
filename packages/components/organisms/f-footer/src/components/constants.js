export const confianzaUrl = 'https://www.confianzaonline.es/empresas/justeatspain.htm';
export const confianzaScreenReaderText = 'Accede a Confianza Online y comprueba nuestro sello de calidad en Internet';
