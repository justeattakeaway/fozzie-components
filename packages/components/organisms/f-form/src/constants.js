export const FORM_EVENTS = {
    fieldUpdated: 'updated',
    submitting: 'form-submitting'
};

export const PROP_VALIDATION_MESSAGES = {
    requiredFormFields: '`formFields` Not found in `formData` prop, please add an array of `formFields`',
    invalidFormFields: '`formFields` are invalid, please add an array of formFields',
    requiredNameProperty: 'Each field in `formFields` should contain "name" property',
    requiredTranslationsProperty: 'Each field in `formFields` should contain "translations" property',
    requiredLabelProperty: 'Each field in `formFields` should contain "translations.label" property'
};

export const DEFAULT_BUTTON_TEXT = 'Submit';
