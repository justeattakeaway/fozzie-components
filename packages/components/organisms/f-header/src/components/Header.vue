<template>
    <header
        :data-theme="theme"
        :class="[
            $style['c-header'],
            headerBackgroundClass,
            transparentBackgroundClasses,
            { [$style['c-header--navInView']]: mobileNavIsOpen }
        ]"
        data-test-id='header-component'>
        <skip-to-main
            v-if="showSkipLink"
            :text="copy.skipToMainContentText"
            :transparent-bg="showTransparentHeader" />

        <div :class="$style['c-header-container']">
            <logo
                :theme="theme"
                :company-name="copy.companyName"
                :is-logo-disabled="isLogoLinkDisabled"
                :logo-gtm-label="copy.logo.gtm"
                :header-background-theme="headerBackgroundTheme"
                :is-open="mobileNavIsOpen" />

            <navigation
                :copy="copy"
                :show-delivery-enquiry="showDeliveryEnquiryWithContent"
                :show-offers-link="showOffersLinkWithContent"
                :show-help-link="showHelpLink"
                :error-log="errorLog"
                :user-info-prop="userInfoProp"
                :user-info-url="userInfoUrl"
                :order-count-url="orderCountUrl"
                :is-order-count-supported="isOrderCountSupported"
                :header-background-theme="headerBackgroundTheme"
                :show-login-info="showLoginInfo"
                :show-country-selector="showCountrySelector"
                @onMobileNavToggle="mobileNavToggled" />
        </div>
    </header>
</template>

<script>
import { globalisationServices } from '@justeat/f-services';
import Logo from './Logo.vue';
import SkipToMain from './SkipToMain.vue';
import { tenantConfigs } from '../tenants';
import Navigation from './Navigation.vue';

export default {
    name: 'VueHeader',

    components: {
        Logo,
        SkipToMain,
        Navigation
    },

    props: {
        locale: {
            type: String,
            default: ''
        },

        errorLog: {
            type: [Function, Boolean],
            default: false
        },

        headerBackgroundTheme: {
            type: String,
            default: 'white'
        },

        isOrderCountSupported: {
            type: Boolean,
            default: true
        },

        isLogoLinkDisabled: {
            type: Boolean,
            default: false
        },

        orderCountUrl: {
            type: String,
            default: '/api/analytics/ordercount'
        },

        showDeliveryEnquiry: {
            type: Boolean,
            default: true
        },

        showLoginInfo: {
            type: Boolean,
            default: true
        },

        showHelpLink: {
            type: Boolean,
            default: true
        },

        showOffersLink: {
            type: Boolean,
            default: false
        },

        showSkipLink: {
            type: Boolean,
            default: true
        },

        userInfoProp: {
            type: [Object, Boolean],
            default: false
        },

        userInfoUrl: {
            type: String,
            default: '/api/account/details'
        },

        showCountrySelector: {
            type: Boolean,
            default: false
        }
    },

    data () {
        const locale = globalisationServices.getLocale(tenantConfigs, this.locale, this.$i18n);
        const localeConfig = tenantConfigs[locale];
        const mobileNavIsOpen = false;

        return {
            copy: { ...localeConfig },
            mobileNavIsOpen
        };
    },

    computed: {
        theme () {
            return globalisationServices.getTheme(this.locale);
        },

        showDeliveryEnquiryWithContent () {
            return this.copy.deliveryEnquiry && this.showDeliveryEnquiry;
        },

        showTransparentHeader () {
            return this.headerBackgroundTheme === 'transparent' && !this.mobileNavIsOpen;
        },

        showOffersLinkWithContent () {
            return this.copy.offers && this.showOffersLink;
        },

        headerBackgroundClass () {
            return this.headerBackgroundTheme === 'highlight' ? this.$style['c-header--highlightBg'] : '';
        },

        transparentBackgroundClasses () {
            return this.showTransparentHeader ? `${this.$style['c-header--transparent']} ${this.$style['c-header--gradient']}` : '';
        }
    },

    methods: {
        mobileNavToggled (value) {
            this.mobileNavIsOpen = value;
        }
    }
};
</script>

<style lang="scss" module>
// when the off-screen navigation is active (on mobile), it fixes to the top of the screen.
// this stops the content being forced upwards when this happens (preventing slight visual glitch)
html:global(.is-navInView) {
    .c-header {
        @include media('<=mid') {
            position: fixed;
            top: 0;
            left: 0;
            width: 100%;
            z-index: zIndex(high);
        }
    }
}

.c-header {
    background-color: $header-bg;
    min-width : 300px;
    position: relative;
    z-index: zIndex(mid);
    box-shadow: $header-box-shadow;

    // Styles for a sticky header on mobile
    @include media('<=mid') {
        &.is-sticky {
            left: 0;
            top: -60px;
            position: fixed;
        }

        &.is-sticky-scrollingUp {
            top: 0;
        }
    }
}

    // Adds a border to the header to separate it from the
    // main content at all widths
    .c-header--bordered {
        box-shadow: $header-box-shadow;
    }

    .c-header--transparent {
        background-color: transparent;
        box-shadow: none;
        position: absolute;
        width: 100%;
    }

    .c-header--gradient {
        &:before {
            content: '';
            height: $header--transparent-gradient;
            opacity: $header--transparent-opacity;
            background-image: linear-gradient(to top, transparent, $header--transparent-gradient-color);
            pointer-events: none;
            position: absolute;
            top: 0;
            left: 0;
            width: 100%;
        }
    }

    .c-header--highlightBg {
        background-color: $color-primary;
        min-height: 88px;
    }

    .c-header-container {
        width: 100%;
        max-width: #{$layout-max-width}px;
        margin: 0 auto;
        padding-left: #{$layout-margin}px;
        padding-right: #{$layout-margin}px;
        position: relative;
        min-height: $header-height--narrow;

        @include media('<wide') {
            padding-left: #{$layout-margin--mid}px;
            padding-right: #{$layout-margin--mid}px;
        }

        @include media('>mid') {
            display: flex;
            min-height: $header-height;
        }

        @include media('<narrow') {
            padding-left: #{$layout-margin--narrow}px;
            padding-right: #{$layout-margin--narrow}px;
        }
    }

    // Header button Styling
    // Example â€“ searchWeb filter button at narrow views
    .c-header-button {
        top: 0;
        right: 0;
        border: 0;
        padding: 0;
        line-height: 1;
        background: none;
        appearance: none;
        position: absolute;
        width: $header-button--width;
        height: $header-button--height;
        z-index: zIndex(belowHighest);

        .is-sticky & {
            top: -#{$header-button--height};
        }
    }

    .c-header-buttonIcon {
        width: 28px;
        height: 15px;
        display: inline-block;

        svg {
            fill: $header-buttonIcon-color;
        }
    }

    .c-header-buttonCount {
        top: 0;
        right: 0;
        min-width: 16px;
        padding: 1px 3px 0;
        text-align: center;
        border-radius: 8px;
        position: absolute;
        @include font-size(caption, false);
        color: $header-buttonCount-color;
        background: $header-buttonCount-bg;
        border: 1px solid $header-buttonCount-borderColor;
    }

    .c-header-button--primary {
        display: block;
        width: 40px;
        padding-right: spacing();
    }

    .c-header-button--secondary {
        right: 40px;
    }
</style>
