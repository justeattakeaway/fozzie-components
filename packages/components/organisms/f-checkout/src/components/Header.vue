<template>
    <div
        data-test-id="checkoutHeader-component"
        :class="$style['c-checkoutHeader']">
        <div
            v-if="isLoggedIn"
            data-test-id="user-checkoutHeader">
            <h2 data-test-id="user-title">
                {{ $t('checkoutHeader.user.title', { name }) }}
            </h2>

            <p :class="$style['c-checkoutHeader-loginLink']">
                <a
                    class="o-link--bold o-link--noDecoration"
                    :href="loginUrl"
                    data-test-id="switch-user-link"
                    @click="onVisitLoginPage">
                    {{ $t('checkoutHeader.user.switchUser', { name }) }}
                </a>
            </p>
        </div>

        <div
            v-else
            data-test-id="guest-checkoutHeader">
            <h2 :class="$style['c-checkoutHeader-title']">
                {{ $t('checkoutHeader.guest.loginTitle') }}
            </h2>

            <f-button
                button-type="primary"
                button-size="large"
                is-full-width
                :class="$style['c-checkoutHeader-loginButton']"
                data-test-id="guest-login-button"
                :href="loginUrl"
                @click.native="onVisitLoginPage">
                {{ $t('checkoutHeader.guest.loginButton') }}
            </f-button>

            <div
                :class="$style['c-checkoutHeader-option']">
                <span>{{ $t('checkoutHeader.guest.option') }}</span>
            </div>

            <h2 :class="$style['c-checkoutHeader-title']">
                {{ $t('checkoutHeader.guest.guestTitle') }}
            </h2>

            <p
                :class="$style['c-checkoutHeader-confirmation']"
                data-test-id="service-type-confirmation">
                {{ $t('checkoutHeader.guest.confirmation', { serviceType }) }}
            </p>
        </div>
    </div>
</template>

<script>
import FButton from '@justeat/f-button';
import '@justeat/f-button/dist/f-button.css';
import { mapState } from 'vuex';
import EventNames from '../event-names';
import { VUEX_CHECKOUT_MODULE } from '../constants';
import loggerMixin from '../mixins/logger.mixin';

export default {
    components: {
        FButton
    },

    mixins: [loggerMixin],

    props: {
        loginUrl: {
            type: String,
            required: true
        }
    },

    computed: {
        ...mapState(VUEX_CHECKOUT_MODULE, [
            'customer',
            'isLoggedIn',
            'serviceType'
        ]),

        name () {
            return (this.customer.firstName.charAt(0).toUpperCase() + this.customer.firstName.slice(1));
        }
    },

    methods: {
        onVisitLoginPage () {
            this.$emit(EventNames.CheckoutVisitLoginPage);

            this.logInvoker({
                message: 'Consumer Visit Login Page',
                logMethod: this.$logger.logInfo
            });
        }
    }
};
</script>

<style lang="scss" module>
.c-checkoutHeader {
    text-align: center;
}

.c-checkoutHeader-title {
    margin-top: 0;
    margin-bottom: spacing(x0.5);
    @include font-size('heading-s');
}

.c-checkoutHeader-loginButton {
    margin-top: spacing(x2);
}

.c-checkoutHeader-confirmation {
    margin-top: spacing(x0.5);
    margin-bottom: 0;
    @include font-size('body-s');
    font-weight: $font-weight-bold;
    color: $grey--dark;
}

.c-checkoutHeader-option {
    display: block;
    margin-top: spacing(x2);
    margin-bottom: spacing(x2);
    text-transform: uppercase;
    font-weight: 300;
    overflow: hidden;

    > span {
        @include font-size('caption');
        position: relative;
        display: inline-block;

        &:before {
            right: 100%;
            margin-right: 60px;
        }

        &:after {
            left: 100%;
            margin-left: 60px;
        }

        &:before,
        &:after {
            content: '';
            position: absolute;
            top: 50%;
            width: 9999px;
            height: 1px;
            background: $grey--mid;
        }
    }
}

.c-checkoutHeader-loginLink {
    text-align: center;
}
</style>
