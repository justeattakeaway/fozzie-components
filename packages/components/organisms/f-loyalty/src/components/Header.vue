<template>
    <div :class="$style['c-loyaltyHeader']">
        <div :class="$style['c-loyaltyHeader-container']">
            <bread-crumbs
                :links="links"
                :router-links="true" />
            <media-element
                :class="$style['c-loyaltyHeader-media']"
                :text-size="fontSize"
                :title="$t('header.title')"
                :text="$t('header.text')"
                :image-url="getImageBasedOnScreenSize"
                image-align="center"
                :flex="flexLayout"
                :alt="$t('header.alt')"
            />
        </div>
    </div>
</template>

<script>
import BreadCrumbs from '@justeat/f-breadcrumbs';
import MediaElement from '@justeat/f-media-element';
import '@justeat/f-media-element/dist/f-media-element.css';
import '@justeat/f-breadcrumbs/dist/f-breadcrumbs.css';
import debounce from 'lodash.debounce';
import { FONT_SIZE } from '@justeat/f-media-element/src/constants';

const REQUIRED_WINDOW_WIDTH = 600;
const TEMP_IMG_PATH = 'https://d30v2pzvrfyzpo.cloudfront.net/b/hw/img/decoration/';

export default {
    name: 'StampcardsHeader',

    components: {
        BreadCrumbs,
        MediaElement
    },

    data: () => ({
        links:  [
            {
                name: 'Home',
                url: '/'
            },
            {
                name: 'For You',
                url: '/offers'
            },
            {
                name: 'StampCards',
                url: '/offers/stamp-cards'
            }
        ],
        fontSize: FONT_SIZE.LG,
        windowWidth: REQUIRED_WINDOW_WIDTH
    }),

    computed: {
        /**
         * Returns the image based on the current window size
         * @returns {string}
         */
        getImageBasedOnScreenSize () {
            return this.windowWidth >= REQUIRED_WINDOW_WIDTH
                ? `${TEMP_IMG_PATH}stampcards-header-${this.$t('percentage')}.svg`
                : `${TEMP_IMG_PATH}stampcards-header-small.svg`;
        },

        /**
         * Sets the media element font size based on the current screen size
         * @returns {string}
         */
        getFontSizeBasedOnScreenSize () {
            return this.windowWidth >= REQUIRED_WINDOW_WIDTH
                ? 'xl' : 'md';
        },

        flexLayout () {
            return {
                default: {
                    column: false,
                    reverse: false
                }
            };
        }
    },

    destroyed () {
        if (process.browser) window.removeEventListener('resize', this.handleResize);
    },

    mounted () {
        if (process.browser) window.addEventListener('resize', debounce(this.handleResize, 300));
        this.handleResize();
    },

    methods: {
        /**
         * Handles the window resize callback
         */
        handleResize () {
            this.windowWidth = process.browser && window.innerWidth;
        }
    }
};
</script>

<style lang="scss" module>

$responsive-mobileViewBreakpoint: '<=narrowMid';

.c-loyaltyHeader-container {
    width: 100%;
    max-width: #{$layout-max-width}px;
    margin: 0 auto;
    padding: #{$layout-margin}px;

    @include media('<mid') {
        padding: #{$layout-margin--mid}px;
    }

    @include media('<narrow') {
        padding: #{$layout-margin--narrow}px;
    }
}

.c-loyaltyHeader {
    padding: spacing() 0 spacing(x6) 0;
    background-color: $color-container-default;
    border-bottom: 1px solid $color-grey-20;
}

.c-loyaltyHeader-media {
    @include media($responsive-mobileViewBreakpoint) {
        margin-top: spacing();
    }
}
</style>
