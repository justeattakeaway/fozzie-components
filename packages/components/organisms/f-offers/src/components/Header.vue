<template>
    <div :class="$style['c-offerHeader-wrapper']">
        <div :class="$style['c-offerHeader-bg']" />
        <div :class="$style['c-offersHeader-container']">
            <media-element
                :class="$style['c-offersHeader-media']"
                :text-size="fontSize"
                :title="title"
                :text="subtitle"
                image-url="https://d30v2pzvrfyzpo.cloudfront.net/b/hw/img/decoration/oi_header-image_v2.svg"
                :image-align="imageAlign"
                :flex="flexLayout"
            />
        </div>
    </div>
</template>

<script>
import { mapGetters } from 'vuex';
import MediaElement from '@justeat/f-media-element';
import '@justeat/f-media-element/dist/f-media-element.css';
import { ALIGN, FONT_SIZE } from '@justeat/f-media-element/src/constants';
import { VUEX_MODULE_NAMESPACE_OFFERS } from '../store/types';

export default {
    name: 'OffersHeader',

    components: {
        MediaElement
    },

    data: () => ({
        fontSize: FONT_SIZE.LG,
        imageAlign: ALIGN.RIGHT
    }),

    computed: {
        ...mapGetters(VUEX_MODULE_NAMESPACE_OFFERS, [
            'isAuthenticated',
            'friendlyName'
        ]),

        flexLayout () {
            return {
                default: {
                    column: false,
                    reverse: false
                },
                modifier: {
                    rule: [
                        '<=',
                        this.isAuthenticated ? 'narrowMid' : 'mid'
                    ],
                    column: true,
                    reverse: true
                }
            };
        },

        title () {
            if (this.isAuthenticated) {
                return this.friendlyName ? `${this.friendlyName},\n ${this.$t('authenticated.header.title')}` :
                    this.$t('authenticated.header.title');
            }
            return this.$t('unauthenticated.header.title');
        },

        subtitle () {
            return this.isAuthenticated ? this.$t('authenticated.header.subtitle')
                : this.$t('unauthenticated.header.subtitle');
        }
    }
};
</script>

<style lang="scss" module>

$header-height: 275px;

.c-offerHeader-wrapper {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    position: relative;
    overflow: hidden;
    height: $header-height;
    background-color: $color-support-brand-02;

    & :global(.c-mediaElement-img--override) {
        width: 200px;
        position: absolute;
        top: -30px;

        @include media('<mid') {
            top: -35px;
            right: -10px;
        }

        @include media('>=mid') {
            width: auto;
            position: relative;
        }

        @include media('>=wide') {
            width: auto;
            position: relative;
            top: -5px;
        }
    }
}

.c-offerHeader-bg {
    position: absolute;
    top: 0;
    bottom: 0;
    right: 0;
    left: 0;
    width: 100%;
    height: 40%;
    background-color: $color-content-brand;
    transform: skewY(6deg);
    transform-origin: top right;

    @include media('>=narrow') {
        height: 45%;
    }

    @include media('>=narrowMid') {
        height: 47.5%;
    }

    @include media('>=mid') {
        height: 60%;
    }

    @include media('>=wide') {
        height: 70%;
    }

    @include media('>=huge') {
        height: 80%;
    }
}

.c-offersHeader-media {
    padding-top: spacing(x10);
    position: relative;
    z-index: 1;
    top: 0;
}

.c-offersHeader-container {
    width: 100%;
    max-width: #{$layout-max-width}px;
    margin: 0 auto;
    padding-left: #{$layout-margin}px;
    padding-right: #{$layout-margin}px;

    @include media('<mid') {
        padding-left: #{$layout-margin--mid}px;
        padding-right: #{$layout-margin--mid}px;
    }

    @include media('<narrow') {
        padding-left: #{$layout-margin--narrow}px;
        padding-right: #{$layout-margin--narrow}px;
    }
}
</style>
