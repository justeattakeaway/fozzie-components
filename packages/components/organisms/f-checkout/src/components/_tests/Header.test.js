import Vuex from 'vuex';
import { VueI18n } from '@justeat/f-globalisation';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Header from '../Header.vue';
import {
    i18n, defaultCheckoutState, createStore, $logger
} from './helpers/setup';
import { CHECKOUT_METHOD_COLLECTION, CHECKOUT_METHOD_DELIVERY } from '../../constants';
import EventNames from '../../event-names';

const localVue = createLocalVue();

localVue.use(VueI18n);
localVue.use(Vuex);

describe('Header', () => {
    allure.feature('Checkout-Header');

    const propsData = {
        loginUrl: 'http://localhost/login'
    };

    it('should be defined', () => {
        // Arrange
        const wrapper = shallowMount(Header, {
            store: createStore(),
            i18n,
            localVue,
            propsData
        });

        // Assert
        expect(wrapper).toBeDefined();
    });

    describe('computed ::', () => {
        describe('isLoggedIn ::', () => {
            describe('when set to true', () => {
                let wrapper;

                beforeEach(() => {
                    wrapper = shallowMount(Header, {
                        store: createStore({ ...defaultCheckoutState, isLoggedIn: true }),
                        i18n,
                        localVue,
                        propsData
                    });
                });

                it('should render the authenticated user header', () => {
                    // Arrange
                    const userHeader = wrapper.find('[data-test-id="user-header"]');

                    // Assert
                    expect(userHeader).toBeDefined();
                });

                it('should show the link to switch users', () => {
                    // Arrange
                    const loginLink = wrapper.find("[data-test-id='switch-user-link']");

                    // Assert
                    expect(loginLink).toBeDefined();
                    expect(loginLink.text()).toBe(`Not ${defaultCheckoutState.customer.firstName}? Click here`);
                });
            });

            describe('when set to false', () => {
                let wrapper;

                beforeEach(() => {
                    wrapper = shallowMount(Header, {
                        store: createStore({ ...defaultCheckoutState, isLoggedIn: false }),
                        i18n,
                        localVue,
                        propsData
                    });
                });

                it('should render the guest header', () => {
                    // Arrange
                    const guestHeader = wrapper.find('[data-test-id="guest-header"]');

                    // Assert
                    expect(guestHeader).toBeDefined();
                });

                it('should show the button to register or login', () => {
                    // Arrange
                    const loginButton = wrapper.find("[data-test-id='guest-login-button']");

                    // Assert
                    expect(loginButton).toBeDefined();
                });
            });
        });

        describe('name ::', () => {
            it('should capitalize `firstName` data', async () => {
                // Act
                const wrapper = shallowMount(Header, {
                    store: createStore({
                        ...defaultCheckoutState,
                        customer: {
                            ...defaultCheckoutState.customer,
                            firstName: 'joe'
                        }
                    }),
                    i18n,
                    localVue,
                    propsData
                });

                const { name } = wrapper.vm;

                // Assert
                expect(name).toEqual('Joe');
            });
        });

        describe('serviceType ::', () => {
            it.each([
                [CHECKOUT_METHOD_DELIVERY],
                [CHECKOUT_METHOD_COLLECTION]
            ])('should update serviceType confirmation with the %s service type', serviceType => {
                // Arrange
                const wrapper = shallowMount(Header, {
                    store: createStore({
                        ...defaultCheckoutState,
                        isLoggedIn: false,
                        serviceType
                    }),
                    i18n,
                    localVue,
                    propsData
                });

                // Act
                const confirmation = wrapper.find("[data-test-id='service-type-confirmation']");
                const received = `Please confirm your ${serviceType} details`;

                // Assert
                expect(confirmation.text()).toEqual(received);
            });
        });
    });

    describe('methods ::', () => {
        describe('onVisitLoginPage ::', () => {
            it('should emit the `VisitLoginPage` event and call `logInfo` when switch user link is clicked.', () => {
                // Arrange
                const wrapper = shallowMount(Header, {
                    store: createStore({ ...defaultCheckoutState, isLoggedIn: true }),
                    i18n,
                    localVue,
                    propsData,
                    mocks: {
                        $logger
                    }
                });

                // Act
                const loginLink = wrapper.find("[data-test-id='switch-user-link']");
                loginLink.trigger('click');

                // Assert
                expect(wrapper.emitted(EventNames.CheckoutVisitLoginPage).length).toBe(1);
                expect($logger.logInfo).toHaveBeenCalled();
            });

            it('should emit the `VisitLoginPage` event when guest login button is clicked.', () => {
                // Arrange
                const wrapper = shallowMount(Header, {
                    store: createStore({ ...defaultCheckoutState, isLoggedIn: false }),
                    i18n,
                    localVue,
                    propsData,
                    mocks: {
                        $logger
                    }
                });

                // Act
                const loginLink = wrapper.find("[data-test-id='guest-login-button']");
                loginLink.trigger('click');

                // Assert
                expect(wrapper.emitted(EventNames.CheckoutVisitLoginPage).length).toBe(1);
                expect($logger.logInfo).toHaveBeenCalled();
            });
        });
    });
});
