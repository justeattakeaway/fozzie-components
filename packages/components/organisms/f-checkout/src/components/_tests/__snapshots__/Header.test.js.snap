// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header computed :: isLoggedIn :: when set to false should render the guest header 1`] = `
<div data-test-id="guest-checkoutHeader">
  <h2>
    How do you want to continue?
  </h2>
  <f-button-stub buttontype="primary" buttonsize="large" isfullwidth="true" actiontype="button" data-test-id="guest-login-button" href="http://localhost/login">
    Log in or sign up
  </f-button-stub>
  <div><span>or</span></div>
  <h2>
    Checkout as a guest
  </h2>
  <p data-test-id="service-type-confirmation">
    Please confirm your delivery details
  </p>
</div>
`;

exports[`Header computed :: isLoggedIn :: when set to true should render the authenticated user header 1`] = `
<div data-test-id="user-checkoutHeader">
  <h2 data-test-id="user-title">
    John, confirm your details
  </h2>
  <p><a href="http://localhost/login" data-test-id="switch-user-link">
      Not John? Click here
    </a></p>
</div>
`;
